File: src/main/java/xyz/deseteral/lebkuchenfm/domain/PingCommandProcessor.java
package xyz.deseteral.lebkuchenfm.domain;

import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Optional;

@Component
class PingCommandProcessor implements CommandProcessor {
    @Override
    public CommandProcessingResponse process(List<String> args) {
        return new CommandProcessingResponse("pong");
    }

    @Override
    public String getKey() {
        return "ping";
    }

    @Override
    public Optional<String> getShortKey() {
        return Optional.empty();
    }

    @Override
    public String getHelpMessage() {
        return "Ping pongs you";
    }
}

File: src/main/kotlin/xyz/deseteral/lebkuchenfm/domain/PingCommandProcessor.kt
package xyz.deseteral.lebkuchenfm.domain

import org.springframework.stereotype.Component

@Component
internal class PingCommandProcessor : CommandProcessor {
    override val key: String
        get() = "ping"

    override val shortKey: String?
        get() = null

    override val helpMessage: String
        get() = "Ping pongs you"

    override fun process(args: List<String>): CommandProcessingResponse {
        return CommandProcessingResponse("pong")
    }
}
